packages/labs/signals/src/lib/watch.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {DirectiveResult, Part, directive} from 'lit/directive.js';
+import {AsyncDirective} from 'lit/async-directive.js';
+import {Signal} from 'signal-polyfill';
+import {SignalWatcher} from './signal-watcher.js';
+
+export class WatchDirective<T> extends AsyncDirective {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private __host?: SignalWatcher;
+  __signal?: Signal.State<T> | Signal.Computed<T>;
+  private __watcher = new Signal.subtle.Watcher(() => {
+    this.__host?.updateWatch(this as WatchDirective<unknown>);
+  });
+
+  commmit() {
+    this.setValue(Signal.subtle.untrack(() => this.__signal?.get()));
+  }
+
+  // @ts-expect-error: signal is unused, but the name appears in the signature
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  render(signal: Signal.State<T> | Signal.Computed<T>): T {
+    return undefined as T;
+  }
+
+  override update(
+    part: Part,
+    [signal]: [signal: Signal.State<T> | Signal.Computed<T>]
+  ) {
+    this.__host ??= part.options?.host as SignalWatcher;
+    if (signal !== this.__signal) {
+      if (this.__signal !== undefined) {
+        this.__watcher.unwatch(this.__signal);
+      }
+      if (signal !== undefined) {
```

Comment:
When is this false? If it's ever false won't line 49 throw?

Reply:
rewritten